state-chain/pallets/cf-elections/src/electoral_systems/mocks.rs

```diff
@@ -96,16 +97,45 @@ where
 
 		TestContext { setup, electoral_access }
 	}
+
+	// We may want to test initialisation of elections within on finalise, so *don't* want to
+	// initialise an election in the utilities.
+	pub fn build(self) -> TestContext<ES> {
+		TestContext {
+			setup: self.clone(),
+			electoral_access: MockAccess::<ES>::new(
+				self.unsynchronised_state,
+				self.unsynchronised_settings,
+				self.electoral_settings,
+			),
+		}
+	}
 }
 
 impl<ES: ElectoralSystem> TestContext<ES> {
 	/// Based on some authority count and votes, evaluate the consensus and the final state.
 	#[allow(clippy::type_complexity)]
 	#[track_caller]
 	pub fn expect_consensus(
+		self,
+		consensus_votes: ConsensusVotes<ES>,
+		expected_consensus: Option<ES::Consensus>,
+	) -> Self {
+		Self::expect_consensus_by(
+			self,
+			consensus_votes,
+			expected_consensus,
+			|new_consensus, expected_consensus| {
+				assert_eq!(new_consensus, expected_consensus);
+			},
+		)
+	}
+
+	pub fn expect_consensus_by<C: FnOnce(Option<ES::Consensus>, Option<ES::Consensus>)>(
 		self,
 		mut consensus_votes: ConsensusVotes<ES>,
 		expected_consensus: Option<ES::Consensus>,
+		check: C,
```

Comment:
This function signature is a bit odd: we pass in the expected_consensus, but we also pass in a closure that receives the expected consensus. 

We could instead just capture the expected consensus in the closure?

Reply:
Just going to delete it since using a BTreeSet makes this redundant